packages/side-nav/src/SideNavItem.tsx

```diff
@@ -180,34 +163,39 @@ export interface SideNavItemProps {
  @param props.as When provided, the component will be rendered as the component or html tag indicated
  *  by this prop. Other additional props will be spread on the anchor element.
  */
-
 const SideNavItem: ExtendableBox<
   SideNavItemProps & { ref?: React.Ref<any> },
   'button'
 > = React.forwardRef((props: SideNavItemProps, forwardRef) => {
   const {
-    active = false,
+    active: activeProp,
     disabled = false,
     ariaCurrentValue = AriaCurrentValue.Page,
     className,
     children,
+    onClick: onClickProp,
+    glyph,
+    ...rest
   } = props;
+  const { usingKeyboard: showFocus } = useUsingKeyboardContext();
+  const active = typeof activeProp === 'boolean' && activeProp;
 
-  const rest = omit(props, [
-    'active',
-    'disabled',
-    'ariaCurrentValue',
-    'className',
-    'children',
-  ]);
+  const onClick = disabled
+    ? (e: React.MouseEvent) => {
```

Comment:
Why do we need to do this specifically when the item is disabled? (As opposed to just not calling `onClickProp()`

Reply:
Yes – we're removing the actual `disabled` attribute usage so that screen reader users can focus on the disabled item. That means we have to manually prevent the click listeners.